### ISSUE TITLE
Multiple warnings with valac 0.39.7

### ISSUE THREAD

**Issue by martinkg** [tiliado/diorite#20]
Hi,

when compiling diorite-4.10.0 on Fedora 28, i get this errors:

```
Compilation succeeded - 5 warning(s)


** (valac:52363): CRITICAL **: 22:53:14.847: vala_data_type_get_actual_type: assertion 'self != NULL' failed
/home/martin/rpmbuild/BUILD/diorite-4.10.0/src/glib/Dbus.vala:80.2-80.22: warning: unreachable code detected
	assert_not_reached();
	^^^^^^^^^^^^^^^^^^^^^
/home/martin/rpmbuild/BUILD/diorite-4.10.0/src/glib/SocketChannel.vala:69.11-71.3: warning: unreachable catch clause detected
/home/martin/rpmbuild/BUILD/diorite-4.10.0/src/glib/Storage.vala:39.2-39.29: warning: Type `GLib.File[]' can not be used for a GLib.Object property
	public GLib.File[] data_dirs
	^^^^^^^^^^^^^^^^^^^^^^^^^^^^

** (valac:52363): CRITICAL **: 22:53:15.431: vala_gir_writer_skip_implicit_params: assertion 'type != NULL' failed

** (valac:52363): CRITICAL **: 22:53:15.431: vala_gir_writer_write_param_or_return: assertion 'type != NULL' failed

** (valac:52363): CRITICAL **: 22:53:15.431: vala_gir_writer_write_implicit_params: assertion 'type != NULL' failed
/home/martin/rpmbuild/BUILD/diorite-4.10.0/src/glib/Dbus.vala:25.1-25.18: warning: Namespace Dbus does not have a GIR namespace and version annotation
namespace Drt.Dbus {
^^^^^^^^^^^^^^^^^^
glib-2.0.vapi:1508.1-1508.14: warning: Namespace GLib does not have a GIR namespace and version annotation
namespace GLib {
^^^^^^^^^^^^^^
```

I can not go back to a older valac version.

**Comment by fenryxo** [comment]
Hello @martinkg, thanks for reporting the issue. However, I cannot see any errors but a few warnings and a successful compilation. I'll keep this ticket open, because I should get rid of the warning.

If Diorite still fails to build on Fedora, please create a new ticket, attach a complete build log as a text file. If you applied any patches, attach them too.

![github_attach](https://user-images.githubusercontent.com/853706/37451617-58b233c8-2832-11e8-932b-a562842ec5ca.png)

**Comment by fenryxo** [comment]
There are tons of `warning: Type XXX can not be used for a GLib.Object property`.

**Comment by fenryxo** [comment]
a lot of compilation warnings on Fedora 28 with diorite-4.10.0 and vala-0.40.0
[warnings-f28.txt](https://github.com/tiliado/diorite/files/1831195/warnings-f28.txt)

**Comment by fenryxo** [comment]
https://github.com/tiliado/diorite/tree/vala0.40 should contain less Valac warnings.

**Comment by martinkg** [comment]
compiled with the las git commit ec98ed1
[diorite-4.10-warnings-f28.txt](https://github.com/tiliado/diorite/files/1841403/diorite-4.10-warnings-f28.txt)

**Comment by martinkg** [comment]
looks now much better with git31408fd
[diorite-4.10-warnings2-f28.txt](https://github.com/tiliado/diorite/files/1841539/diorite-4.10-warnings2-f28.txt)

**Comment by fenryxo** [comment]
There are only some deprecation warnings left, which can be ignored for now.

**Comment by fenryxo** [comment]
Diorite mater now builds without any warnings with Vala 0.41.91. Moreover, it treats warnings as fatal by default unless you use `./waf configure --no-strict`.

**Comment by fenryxo** [comment]
[Same with Vala 0.40.8 in Fedora 28](https://circleci.com/gh/tiliado/diorite/17).

**Comment by fenryxo** [comment]
[Released in Nuvola 4.13.0](https://medium.com/nuvola-news/nuvola-4-13-released-2fdf80c73742).
